tests/unit/helper/extractMetadata.spec.ts

```diff
@@ -0,0 +1,50 @@
+import { vi } from 'vitest'
+import * as extractMetadata from '../../../src/helper/extractMetadata'
+
+const mockImageData = {
+  byteArray: [],
+  elements: {}
+}
+
+vi.mock('@cornerstonejs/dicom-image-loader', () => ({
+  wadouri: {
+    loadImage: () => ({
+      promise: Promise.resolve({
+        data: mockImageData
+      })
+    })
+  }
+}))
+
+describe('Methods', () => {
+  describe('method: fetchDicomImageData', () => {
+    it('should return dicom image data', async () => {
+      const dicomImageData = await extractMetadata.fetchDicomImageData('wadouri:https://test')
+      expect(dicomImageData).toMatchObject(mockImageData)
+    })
+  })
+
+  describe('method: extractDicomMetaData', () => {
+    it('should return extracted dicom metadata', async () => {
+      const imageData = {
+        string: function (tag: string) {
+          switch (tag) {
+            case 'x00100010':
+              return 'testPatient'
+          }
+        }
+      }
+
+      const dicomMetaData = await extractMetadata.extractDicomMetadata(
+        imageData,
+        ['patientName', '', undefined, null],
+        'en'
+      )
```

Comment:
I think you need to test more inputs. In addition to a standard tag like "patientName" that just returns the value from meta data and empty/undefined value you need to add
- some tag with _formatDate like for example "patientBirthdate" which should return a date value that has been formatted nicely
- some tag with _formatTime like instanceCreationTime which should return a time value that has been formatted nicely
- some tag with _addSOPuids like SOP_ClassUID which should return the SOP class together with the description
You might also want to test what happend if someone passes an invalid tag into this function (I think it should return undefined)